### PR TITLE
fix: fixed memory leak in user policies and dashboard

### PR DESCRIPTION
**Pull request by sidharthjain-zemoso**
# Description
Going from dashboard to user policies and then switching asset group leads to dashboard APIs getting called.
This issue occured because of not unsubscribing the agDomainSubscription in Dashboard and Policies. 

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Chore (no code changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also
list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] My commit message/PR follows the contribution guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

# **Other information**:

List any documentation updates that are needed for the Wiki

### CODE REVIEW

File: webapp/src/app/pacman-features/modules/compliance/compliance-dashboard/compliance-dashboard.component.ts [hunk 1]
```diff
@@ -1402,8 +1401,8 @@ export class ComplianceDashboardComponent implements OnInit, OnDestroy {
 
     ngOnDestroy() {
         try {
-            if (this.assetGroupSubscription) {
-                this.assetGroupSubscription.unsubscribe();
+            if(this.agDomainSubscription){
```
Comment (plyubich) [inline webapp/src/app/pacman-features/modules/compliance/compliance-dashboard/compliance-dashboard.component.ts:1404]: apply formatting on the file.

File: webapp/src/app/pacman-features/modules/compliance/policy-knowledgebase/policy-knowledgebase.component.ts [hunk 2]
```diff
@@ -896,6 +897,9 @@ export class PolicyKnowledgebaseComponent implements OnInit, AfterViewInit, OnDe
 
   ngOnDestroy() {
     try {
+      if(this.agDomainSubscription){
```
Comment (plyubich) [inline webapp/src/app/pacman-features/modules/compliance/policy-knowledgebase/policy-knowledgebase.component.ts:900]: formatting

File: webapp/src/app/post-login-app/default-asset-group/default-asset-group.component.ts [hunk 3]
```diff
@@ -299,6 +301,9 @@ export class DefaultAssetGroupComponent implements OnInit, OnDestroy {
 
   ngOnDestroy() {
     try {
+      if(this.agDomainSubscription){
```
Comment (plyubich) [inline webapp/src/app/post-login-app/default-asset-group/default-asset-group.component.ts:304]: formatting

**Review by plyubich — COMMENTED** [review]
please format affected files

File: webapp/src/app/pacman-features/modules/compliance/compliance-dashboard/compliance-dashboard.component.ts (hunk 1 above)
Comment (sidharthjain-zemoso) [reply webapp/src/app/pacman-features/modules/compliance/compliance-dashboard/compliance-dashboard.component.ts]: Done

**Comment by sidharthjain-zemoso** [comment]
> please format affected files

Applied formatting to affected files. Can you please review again ?

**Review by santhosh-challa — APPROVED** [review]
LGTM

**Pull request merged**